Sources/RTMP/RTMPStream.swift

```diff
@@ -416,7 +416,7 @@ open class RTMPStream: IOStream {
         return metadata
     }
 
-    override public func readyStateWillChange(to readyState: IOStream.ReadyState) {
+    override open func readyStateWillChange(to readyState: IOStream.ReadyState) {
```

Comment:
If the goal is to detect changes in the readyState, I think it would be more iOS-like to provide a dedicated delegate method. Would achieving the requirements be difficult with such an approach?

Reply:
Done, please take a look